extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -412,6 +419,12 @@ public boolean publishSegments(Set<DataSegment> segments, Object commitMetadata)
             throw new ISE("WTF?! Driver attempted to publish invalid metadata[%s].", commitMetadata);
           }
 
+          // Upgrade TaskLocks for all segments
+          for (DataSegment segment: segments) {
+            toolbox.getTaskActionClient().submit(new SetLockCriticalStateAction(segment.getInterval(),
+                                                                                TaskLockCriticalState.UPGRADE));
```

Comment:
@pjain1 Do you mean the exact same interval or just one that contains it? It will have a lock that contains the segment interval, but it might not be the exact same interval.